File: ai/models/channel_summary.py
from typing import Optional
from datetime import datetime
from pydantic import BaseModel
import json
from utils import string_utils


class ChannelSummary(BaseModel):
    id: int
    channelName: Optional[str] = None
    parentId: Optional[int] = None
    parentsPath: Optional[list[int]] = None
    indexName: Optional[str] = None
    knowledge: Optional[bool] = None
    tableName: Optional[str] = None
    taxis: Optional[int] = None
    addDate: Optional[datetime] = None

    @classmethod
    def load_summaries(cls, json_str: str | None) -> list["ChannelSummary"]:
        if json_str is None:
            return []
        return json.loads(json_str, object_hook=cls.model_validate)

    @classmethod
    def json_summaries(cls, summaries: list["ChannelSummary"] | None) -> str:
        if summaries is None:
            return "[]"
        return string_utils.to_json_str(summaries)

File: core/src/SSRAG/Models/ChannelSummary.cs
using System;
using System.Collections.Generic;

namespace SSRAG.Models
{
    public class ChannelSummary : IChannelSummary
    {
        public int Id { get; set; }
        public string ChannelName { get; set; }
        public int ParentId { get; set; }
        public List<int> ParentsPath { get; set; }
        public string IndexName { get; set; }
        public bool Knowledge { get; set; }
        public string TableName { get; set; }
        public int Taxis { get; set; }
        public DateTime? AddDate { get; set; }
    }
}
